pyi.py

```diff
@@ -695,6 +695,10 @@ def _analyse_union(members: Sequence[ast.expr]) -> UnionAnalysis:
     )
 
 
+_ALLOWED_MATH_ATTRIBUTES_IN_DEFAULTS = frozenset(
+    {"math.inf", "math.e", "math.pi", "math.tau"}
```

Comment:
Maybe add `math.nan`.

Reply:
I'm fine to add it if requested. I have rarely seen `nan` in code. Also, I don't `nan` should be considered a constant:

```python
>>> import math
>>> math.nan == math.nan
```

And having `-nan` is a bit weird to me:

```python
>>> -math.inf
-inf
>>> -math.nan
nan
```

One use case for having `nan` as default may be:

```python
def mean(iterable: Iterable[float], default: float = math.nan) -> float:
    sequence = list(iterable)
    if sequence:
        return sum(sequence) / len(sequence)
    return default
```

But I think the code about should explicitly raise an error rather than silently fails to `nan`.

Reply:
We're a linter for stub files, so I don't think it's our place to have an opinion on whether it's a good idea to use nan as a default. Instead, we should consider whether for functions that use `nan` as a default at runtime, there's a reason to stop stubs from using that same default. And I don't see a strong reason why we shouldn't allow `nan` if we already allow other constants from `math`.

Reply:
> We're a linter for stub files, so I don't think it's our place to have an opinion on whether it's a good idea to use nan as a default. Instead, we should consider whether for functions that use `nan` as a default at runtime, there's a reason to stop stubs from using that same default. And I don't see a strong reason why we shouldn't allow `nan` if we already allow other constants from `math`.

That rationale makes sense to me! @XuehaiPan, could you make the change? :)